Source/messaging/MessageUnit.cpp

```diff
@@ -312,7 +314,6 @@ namespace Thunder {
                     Messaging::ConsoleStandardOut::Instance().Close();
                 }
                 Core::Messaging::IStore::Set(nullptr);
-                Core::Messaging::IControl::Iterate(handler);
```

Comment:
Here it also looks like we are getting rid of a `Destroy()` call via iterate on all handles. Basically the whole section above with the `Handler` class definition and its `Handle()` method calling `Destroy()` is no longer used, why is that?

Reply:
You might recall the discussion outside the Github context about the ANNOUNCE-macros and element initially added do not re-appear on open-close-open of MessageUnit?

Reply:
See https://github.com/rdkcentral/Thunder/commit/cdc50b28ec2c975472662643ddf091c66d6bce36

Reply:
I see, but it looks to me that if we drop this Iterate() and thus Destroy(), the other side (e.g. [here](https://github.com/rdkcentral/Thunder/blob/master/Source/messaging/TraceControl.h#L46)) will not be aware that MessageUnit has been closed and its controls should no longer be used

Reply:
You (@VeithMetro) mentioned the use case is not what had been considered in the original design. However, from the tests it can be concluded that is a defect iff allowed. Imho allowing an explicit Close suggests the code flow is allowed. A more recent test illustrates no current issue by re-adding the Destroy(). However, the code is not reached in any of the current tests (anymore). I suggest for now it is best to revert the removal of Destroy() and see what the future brings, or, try to implement a test which deliberately can excite that code point. What is your opinion?

Reply:
I think having a test which deliberately can excite that point in code, and which preferably suits a real use case would be great. Have you maybe tried to run Thunder to see if the Close() is ever executed? E.g. when closing MessageControl or other plugins which use messaging?

Reply:
Agreed, that's why in messaging we make sure that a control will always be Revoked() ([here](https://github.com/rdkcentral/Thunder/blob/200249e3099531df2858c68f6b94f7c2d60c597e/Source/messaging/Control.h#L71)) thus removed from our `_controlList`